source/Nevermore/Advanced/Queryable/QueryProvider.cs

```diff
@@ -86,7 +87,27 @@ public async Task<TResult> ExecuteAsync<TResult>(Expression expression, Cancella
             {
                 var asyncStream = (IAsyncEnumerable<object>)GenericStreamAsyncMethod.MakeGenericMethod(expression.Type)
                     .Invoke(queryExecutor, new object[] { command, cancellationToken });
-                return (TResult)Convert.ChangeType(await FirstOrDefaultAsync(asyncStream, cancellationToken), typeof(TResult));
```

Comment:
To address the issue it should be sufficient to just remove the call to `Convert.ChangeType()`. It's just a remnant of how this method used to work, and is now unnecessary.